test/cli/whole-program/odr3.cpp

```diff
@@ -0,0 +1,6 @@
+// #10431
+#ifdef X
+struct S { int i; }; // cppcheck-suppress unusedStructMember
```

Comment:
shouldn't you add this file `odr3.cpp` to whole-program_test.py in some places?

Reply:
I guess this could become tricky.. but what happens if you have two files..

file1.cpp:
```
#ifdef X
struct S { int i; }; 
#endif
```

file2.cpp:
```
struct S { int i; }; 
```
These configurations are different but not mutually exclusive.

would it make sense to make sure that if file0 is different then all configurations are considered. If the user don't want to consider all configurations in this case then we can recommend that `-D` that matches compilation is used?

Reply:
> shouldn't you add this file `odr3.cpp` to whole-program_test.py in some places?

Absolutely, I must have thought that all files would be checked automatically.